django/contrib/flatpages/forms.py

```diff
@@ -22,6 +22,13 @@ class Meta:
         model = FlatPage
         fields = '__all__'
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if settings.APPEND_SLASH and 'django.middleware.common.CommonMiddleware' in settings.MIDDLEWARE:
```

Comment:
That's an interesting thought. I'll try it out and see what happens. I was originally concerned about how it would impact people changing settings at runtime (which isn't really supported but I know it happens), but I think the chance of people messing with middleware and `APPEND_SLASH` at runtime is pretty remote anyway.